lib/bootstrap.rb

```diff
@@ -0,0 +1,118 @@
+module LogStash
+  module Environment
+    extend self
+
+    # also set the env LOGSTASH_HOME
+    LOGSTASH_HOME = ENV["LOGSTASH_HOME"] = ::File.expand_path(::File.join(__FILE__, "..", ".."))
+
+    BUNDLE_DIR = ::File.join(LOGSTASH_HOME, "vendor", "bundle")
+    GEMFILE_PATH = ::File.join(LOGSTASH_HOME, "Gemfile")
+
+    # @return [String] the ruby version string bundler uses to craft its gem path
+    def gem_ruby_version
+      RbConfig::CONFIG["ruby_version"]
+    end
+
+    # @return [String] major.minor ruby version, ex 1.9
+    def ruby_abi_version
+      RUBY_VERSION[/(\d+\.\d+)(\.\d+)*/, 1]
+    end
+
+    # @return [String] jruby, ruby, rbx, ...
+    def ruby_engine
+      RUBY_ENGINE
+    end
+
+    def logstash_gem_home
+      ::File.join(BUNDLE_DIR, ruby_engine, gem_ruby_version)
+    end
+  end
+
+  module Bootstrap
+    extend self
+
+    def bundler_setup!(options = {})
```

Comment:
In order to enforce clarity I see this code belonging to a bundler module, that then can be used from here when necessary.